--- reiserfsprogs-3.x.0b/lib/misc.c.sec	Wed Jan 24 00:01:23 2001
+++ reiserfsprogs-3.x.0b/lib/misc.c	Wed Jan 24 00:08:39 2001
@@ -249,21 +249,19 @@
 static int is_readonly_dir (char * dir)
 {
     char * name;
-    FILE * f;
+    int fd;
+    char template[1024];
 
-    name = tempnam (dir, 0);
-    if (!name) {	
-	fprintf (stderr, "is_readonly: tempnam failed, think fs is not readonly\n");
-	return 0;
-    }
+    if(dir==NULL)
+	    strcpy(template, "/tmp/testXXXXXX");
+    else
+	    snprintf(template, 1024, "%s/testXXXXXX", dir);
 
-    f = fopen (name, "w");
-    if (f) {
-	unlink (name);
-	free (name);
-	return 0;
+    fd = mkstemp(template);
+    if(fd >= 0) {
+	    close(fd);
+	    return 0;
     }
-    free (name);
     return (errno == EROFS) ? 1 : 0;
 }
 
