diff -urNp -x '*.orig' reiserfsprogs-3.6.27.org/configure.ac reiserfsprogs-3.6.27/configure.ac
--- reiserfsprogs-3.6.27.org/configure.ac	2017-07-25 15:16:14.000000000 +0200
+++ reiserfsprogs-3.6.27/configure.ac	2026-03-03 00:04:00.766490233 +0100
@@ -12,7 +12,7 @@ AM_MAINTAINER_MODE
 dnl We install in /sbin, the utils are to be available on boot
 dnl AC_PREFIX(/)
 
-AM_CONFIG_HEADER(include/config.h)
+AC_CONFIG_HEADERS(include/config.h)
 
 dnl Checks for programs.
 AC_PROG_CC
